lib/MumbleConnection.js

```diff
@@ -554,8 +611,15 @@ MumbleConnection.prototype._processAudio = function() {
             // We got listeners for voice event so do decode.
             var decoded = [];
             for( var f in packets ) {
+                var data;
+                if(this.opusEnabled) {
+                    data = packets[f].decoder.decode( packets[f].frame, this.FRAME_SIZE );
+                }
+                else {
+                    data = packets[f].decoder.decode( packets[f].frame );
+                }
                 decoded.push({
-                    data: packets[f].decoder.decode( packets[f].frame ),
+                    data: data,
                     codec: packets[f].codec,
```

Comment:
The packets have their own codec information. Clients might be sending celt encoded packets when the server decides to force switch to opus (or vice versa). Or the UDP packets might get reordered.

So instead of having global "opusEnabled" flag, we should decide the correct encoder based on the packet codec attribute.

This is why having both encoders ready is probably the way to go.